### PR TITLE
Port to 1.18

### PR DESCRIPTION
**Pull request by Fourmisain**
Did this back in December, just didn't PR it earlier because #30 (which this PR is based on) is still up, now with 1.18.2 coming up I felt like making it visible, so here you go.

This port was way easier than the 1.17 one. Apart from the gradle/loom/java updates, there's really just one commit: https://github.com/benfah/SimpleDrawers/commit/96de1708c52caf577ab9ba2eb358eb79da02fdb4
Fabric API removed the `BlockEntityClientSerializable` because it is now mostly unneeded.
There's a really helpful explanation of it here: https://github.com/FabricMC/fabric/discussions/1779

Basically:
Implement `toInitialChunkDataNbt()` and `toUpdatePacket()` and replace the now removed `BlockEntityClientSerializable.sync()` call with `getChunkManager().markForUpdate()`.

P.S.
You can create a new branch (e.g. by pushing) and redirect this PR to it with the Edit button on the top right.

### CODE REVIEW

**Comment by Fourmisain** [comment]
Added a boot crash fix for e.g. Botania and Create.

 - `ModelLoader.bakedModels` changed it's `ModelIdentifier` type to `Identifier` - which in itself doesn't really cause an issue because of type erasure.

- `ModelMerger.onPostBake` assumed that every model with a `border_type` needs a drawer border and tried to cast `Identifier` to `ModelIdentifier`.

For the latter I chose the simple route of checking the mod id since I'm certain that no mod tries to add custom drawer borders right now. A proper fix would probably need to check the `BorderRegistry` but it wasn't super clear how exactly to go there from just the ModelIdentifier.
